simple/stats/events_importer.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import random
+
+import pandas as pd
+from stats import constants
+from stats.data import Event
+from stats.data import Triple
+from stats.db import Db
+from stats.importer import Importer
+from stats.nodes import Nodes
+from stats.reporter import FileImportReporter
+from util.filehandler import FileHandler
+
+from util import dc_client as dc
+
+# Number of entity IDs that will be sampled to resolved their entity type, if one is not specified by the user.
+# Note that the importer assumes that all entities in a given CSV are all of the same type.
+_SAMPLE_ENTITY_RESOLUTION_SIZE = 5
+
+
+class EventsImporter(Importer):
+  """Imports a single events input file.
+    """
+
+  def __init__(self, input_fh: FileHandler, db: Db,
+               debug_resolve_fh: FileHandler, reporter: FileImportReporter,
+               nodes: Nodes) -> None:
+    self.input_fh = input_fh
+    self.db = db
+    self.debug_resolve_fh = debug_resolve_fh
+    self.reporter = reporter
+    self.nodes = nodes
+    self.input_file_name = self.input_fh.basename()
+    self.config = nodes.config
+    self.entity_type = self.config.entity_type(self.input_file_name)
+    self.ignore_columns = self.config.ignore_columns(self.input_file_name)
+    self.provenance = self.nodes.provenance(self.input_file_name).id
+
+    self.event_type = self.config.event_type(self.input_file_name)
+    assert self.event_type, f"Event type must be specified: {self.input_file_name}"
+
+    self.id_column = self.config.id_column(self.input_file_name)
+
+    self.df = pd.DataFrame()
+    self.debug_resolve_df = None
+
+  def do_import(self) -> None:
+    self.reporter.report_started()
+    try:
+      self._read_csv()
+      self._drop_ignored_columns()
+      self._sanitize_values()
+      self._resolve_entities()
+      self._parse_dates()
+      self._rename_columns()
```

Comment:
maybe do _rename_columns() immediately after _read_csv() and later functions can all use column name (instead of index) for clarity.

Reply:
Ack - will do in a separate PR (see comment above).